src/reducers/weeksReducer.js

```diff
@@ -0,0 +1,18 @@
+import { SET_WEEKS } from "../actions/TYPES";
+
+export default function userReducer(
+  state = {
+    weeks: [],
+  },
+  action
+) {
+  switch (action.type) {
+    case SET_WEEKS:
+      return {
```

Comment:
is there a reason we don't do get_weeks here too. Instead of doing the async function in component we could move it to a handler.